modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java

```diff
@@ -42,16 +66,63 @@ public void pkWithNullableColumns() {
 
     @Test
     public void undefinedColumnsInPrimaryKey() {
-        assertThrows(
-                IgniteException.class,
-                () -> sql("CREATE TABLE T0(ID INT, VAL INT, PRIMARY KEY (ID1, ID0, ID2))"),
-                " Primary key constrain contains undefined columns: [cols=[ID0, ID2, ID1]]"
+        assertThat(
+                assertThrows(
+                        IgniteException.class,
+                        () -> sql("CREATE TABLE T0(ID INT, VAL INT, PRIMARY KEY (ID1, ID0, ID2))")
+                ).getMessage(),
+                containsString("Primary key constrain contains undefined columns: [cols=[ID0, ID2, ID1]]")
         );
     }
 
+    /**
+     * Check invalid colocation columns configuration:
+     * - not PK columns;
+     * - duplicates colocation columns.
+     */
     @Test
-    public void dbg() {
-        sql("CREATE TABLE T0(ID INT, VAL INT, PRIMARY KEY (ID))");
+    public void invalidColocationColumns() {
+        assertThat(
+                assertThrows(
+                        IgniteException.class,
+                        () -> sql("CREATE TABLE T0(ID0 INT, ID1 INT, VAL INT, PRIMARY KEY (ID1, ID0)) COLOCATE (ID0, VAL)")
+                ).getMessage(),
+                containsString("Schema definition error: All colocation columns must be part of primary key")
+        );
+
+        assertThat(
+                assertThrows(
+                        IgniteException.class,
+                        () -> sql("CREATE TABLE T0(ID0 INT, ID1 INT, VAL INT, PRIMARY KEY (ID1, ID0)) COLOCATE (ID1, ID0, ID1)")
```

Comment:
COLOCATE and COLOCATE BY - work both?

Reply:
Yes. I guess it's OK because e.g. `ALTER TABLE ADD` and `ALTER TABLE ADD COLUMN` work both. 
We can remove it now.

Reply:
I think it's ok, let's keep it